internal/archive/archive.go

```diff
@@ -148,6 +195,18 @@ func openUbuntu(options *Options) (Archive, error) {
 		pubKeys: options.PubKeys,
 	}
 
+	if archive.baseURL == "" {
+		archive.baseURL = archiveURL(options.Pro, options.Arch)
+	}
+
+	if options.Pro != "" && archive.creds == nil {
```

Comment:
Same question, how would `archive.creds` be set at this point?

Reply:
Same, changed.